gio/gappinfo.c

```diff
@@ -824,12 +846,68 @@ launch_default_with_portal_async (const char          *uri,
   g_free (real_uri);
 }
 
+static void
+launch_default_with_portal_sync (const char          *uri,
+                                 GAppLaunchContext   *context)
+{
+  GDBusConnection *session_bus;
+  GVariantBuilder opt_builder;
+  GVariant *res = NULL;
+  const char *parent_window = NULL;
+  char *real_uri;
+  GError *error = NULL;
+
+  session_bus = g_bus_get_sync (G_BUS_TYPE_SESSION, NULL, &error);
+  if (session_bus == NULL)
+    {
+      g_task_report_error (context, NULL, NULL, NULL, error);
+      return;
+    }
+
+  if (context && context->priv->envp)
```

Comment:
This is not really my code, but taken from [the original async version of this function](https://github.com/endlessm/glib/pull/26/files/879fd07fb253ce751775219dd3485fb09f2afc6f..144ce1b68341c408f8d8ca06d33f13e58e091377#diff-3edd4452eed63275cb15ca0bb9b22b2fL771), so I'd rather leave it like that for consistency.

I thought about refactoring into a function but it didn't seem worth the effort, wouldn't solve much and would make code less readable, so I just left it as it is.